src/formatDate.js

```diff
@@ -9,6 +9,65 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let year = '';
+  let month = '';
+  let day = '';
+  let string = '';
+  let newDate = '';
+  const sign = toFormat[toFormat.length - 1];
+  const dateParts = date.split(/[.\-/]/);
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    const meaning = fromFormat[i];
+
+    if (meaning === 'YYYY') {
+      year += dateParts[i];
+    }
+
+    if (meaning === 'YY') {
+      year = parseInt(dateParts[i]);
+
+      if (year > 0 && year < 30) {
+        year = '20' + year;
+      } else if (year === 0) {
+        year = '200' + year;
+      } else {
+        year = '19' + year;
+      }
+    }
+
+    if (meaning === 'MM') {
+      month += dateParts[i];
+    }
+
+    if (meaning === 'DD') {
+      day += dateParts[i];
+    }
+  }
```

Comment:
```suggestion
  const newDate = [];
  const objectDate = {};
  const splitDate = date.split(oldSeparator);

  for (let i = 0; i < splitDate.length; i++) {
    objectDate[fromFormat[i]] = splitDate[i];
  }
  
  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
    if (objectDate['YY'] < 30) {
      objectDate['YYYY'] = '20' + objectDate['YY'];
    } else {
      objectDate['YYYY'] = '19' + objectDate['YY'];
    }
  }
```
we can simplify this logic, then just slice the year field if needed, and fill the newDate array with objectDate values